SoftLayer/managers/vs.py

```diff
@@ -579,9 +576,34 @@ def _generate_create_dict(
 
         return data
 
+    def get_network_components(self, data, kwargs, private_subnet, private_vlan, public_subnet, public_vlan):
```

Comment:
I think you could combine this function with `_create_network_components`,  `kwargs.get()` should rerturn `None` if the option wasn't specified. So really you just need to move the Exception where your checking for vlan/subnets and router to `create_network_components` and save having to make another function.

Reply:
Done.